web/src/js/ducks/modes/reverse.ts

```diff
@@ -0,0 +1,139 @@
+import { ModeState, updateMode } from "../modes";
+import {
+    RECEIVE as RECEIVE_OPTIONS,
+    UPDATE as UPDATE_OPTIONS,
+} from "../options";
+
+export const TOGGLE_REVERSE = "TOGGLE_REVERSE";
+export const ERROR_REVERSE = "ERROR_REVERSE";
+export const SET_PROTOCOLS = "SET_PROTOCOLS";
+
+interface ProtocolState {
+    name: string;
+    isSelected: boolean;
+}
+
+interface ReverseState extends ModeState {
+    protocols: ProtocolState[];
+}
+
+export const initialState: ReverseState = {
+    active: false,
+    protocols: [
+        { name: "http", isSelected: false },
+        { name: "https", isSelected: false },
+        { name: "dns", isSelected: false },
+        { name: "http3", isSelected: false },
+        { name: "quic", isSelected: false },
+        { name: "tcp", isSelected: false },
+        { name: "tls", isSelected: false },
+        { name: "udp", isSelected: false },
+        { name: "dtls", isSelected: false },
```

Comment:
Is there a reason why this is not
```javascript
protocol: "http" | "https" | "dns" | ...
```
? 😅

Reply:
yes I don't know why I opted for this solution, it's probably not necessary to have an `isSelected` flag 😅